SC panics with nil pointer dereference

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c67451]

goroutine 420 [running]:
github.com/fluxcd/source-controller/controllers.(*HelmRepositoryReconciler).notify(0xc0003e4660, 0xc0008da340, 0xc000c01040, {{0xc000618c30, 0x26}, {0x28875e0, 0xc0006f6780}, {0xc0002dbc68, 0x3, 0x3}, ...}, ...)
        github.com/fluxcd/source-controller/controllers/helmrepository_controller.go:258 +0x1b1
```
This is coming from https://github.com/fluxcd/source-controller/blob/v0.24.0/controllers/helmrepository_controller.go#L258 where SC in fact doesn't check if `newObj.Status.Artifact.Size` is nil, dereferencing it in any case.

I'm not sure in which situation this field would be nil but I don't see it set in older, working instances of SC, too.

--- a/controllers/helmrepository_controller.go
+++ b/controllers/helmrepository_controller.go
@@ -255,14 +255,17 @@ func (r *HelmRepositoryReconciler) notify(oldObj, newObj *sourcev1.HelmRepositor
 			sourcev1.GroupVersion.Group + "/checksum": newObj.Status.Artifact.Checksum,
 		}
 
-		size := units.HumanSize(float64(*newObj.Status.Artifact.Size))
+		humanReadableSize := "unknown size"
+		if size := newObj.Status.Artifact.Size; size != nil {
+			humanReadableSize = fmt.Sprintf("size %s", units.HumanSize(float64(*size)))
+		}
 
 		var oldChecksum string
 		if oldObj.GetArtifact() != nil {
 			oldChecksum = oldObj.GetArtifact().Checksum
 		}
 
-		message := fmt.Sprintf("stored fetched index of size %s from '%s'", size, chartRepo.URL)
+		message := fmt.Sprintf("stored fetched index of %s from '%s'", humanReadableSize, chartRepo.URL)
 
 		// Notify on new artifact and failure recovery.
 		if oldChecksum != newObj.GetArtifact().Checksum {

--- a/controllers/helmrepository_controller_test.go
+++ b/controllers/helmrepository_controller_test.go
@@ -860,6 +860,15 @@ func TestHelmRepositoryReconciler_notify(t *testing.T) {
 			resErr: errors.New("some error"),
 		},
 		{
+			name:   "new artifact with nil size",
+			res:    sreconcile.ResultSuccess,
+			resErr: nil,
+			newObjBeforeFunc: func(obj *sourcev1.HelmRepository) {
+				obj.Status.Artifact = &sourcev1.Artifact{Revision: "xxx", Checksum: "yyy", Size: nil}
+			},
+			wantEvent: "Normal NewArtifact stored fetched index of unknown size",
+		},
+		{
 			name:   "new artifact",
 			res:    sreconcile.ResultSuccess,
 			resErr: nil,

